docs/DeveloperGuide.md

```diff
@@ -2,37 +2,200 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+Parser class and its functions are largely adapted from Ria's IP and help from regex101 (https://regex101.com/).
 
-## Design & implementation
+Storage class and its functions are largely adapted from Dhanish's IP on Duke.
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Design & implementation
 
+The Patient and PatientList classes are used in conjunction to manage patients, and the list of patients. Each patient has a name,
+a unique ID, a date of birth and gender. The PatientList class holds an ArrayList of Patients and manipulates them accordingly.
+
+At the start of the program, a new PatientList object is instantiated. Through methods in the Storage class, data is read from the
+relevant text files to create Patients that existed prior to the last closure of the program, and then adds these patients to the
+ArrayList in PatientList. This finishes the initial set-up.
+
+### Methods in PatientList class:
+
```

Comment:
Perhaps a sequence diagram could be added to better give an understanding on the methods